src/SME.SGP.Aplicacao/Queries/Notas/ObterNotasPorAlunosAtividadesAvaliativas/ObterNotasPorAlunosAtividadesAvaliativasQueryHandler.cs

```diff
@@ -11,14 +12,28 @@ namespace SME.SGP.Aplicacao
     public class ObterNotasPorAlunosAtividadesAvaliativasQueryHandler : IRequestHandler<ObterNotasPorAlunosAtividadesAvaliativasQuery, IEnumerable<NotaConceito>>
     {
         private readonly IRepositorioNotasConceitosConsulta repositorioNotasConceitos;
-
-        public ObterNotasPorAlunosAtividadesAvaliativasQueryHandler(IRepositorioNotasConceitosConsulta repositorioNotasConceitos)
+        private readonly IRepositorioCache repositorioCache;
+        public ObterNotasPorAlunosAtividadesAvaliativasQueryHandler(IRepositorioNotasConceitosConsulta repositorioNotasConceitos, IRepositorioCache repositorioCache)
         {
             this.repositorioNotasConceitos = repositorioNotasConceitos ?? throw new ArgumentNullException(nameof(repositorioNotasConceitos));
+            this.repositorioCache = repositorioCache ?? throw new ArgumentNullException(nameof(repositorioCache));
         }
         public async Task<IEnumerable<NotaConceito>> Handle(ObterNotasPorAlunosAtividadesAvaliativasQuery request, CancellationToken cancellationToken)
         {
-            return await repositorioNotasConceitos.ObterNotasPorAlunosAtividadesAvaliativasAsync(request.AtividadesAvaliativasId, request.AlunosId, request.ComponenteCurricularId);
+            var nomeChave = $"Atividade-Avaliativa-{request.CodigoTurma}";
+            var atividadesAvaliativasNoCache = await repositorioCache.ObterAsync(nomeChave);
+
+            IEnumerable<NotaConceito> atividadeAvaliativas;
+
+            if (string.IsNullOrEmpty(atividadesAvaliativasNoCache))
+            {
+                atividadeAvaliativas = await repositorioNotasConceitos.ObterNotasPorAlunosAtividadesAvaliativasPorTurmaAsync(request.CodigoTurma);
```

Comment:
Transforme esse trecho aqui em um command, que tu irá usar para manter sempre atualizado o cache quando uma atividade avaliativa for criada novamente na turma, aí tu vai simplesmente fazer assim CriarCacheDeAvaliacaoAvaliativaPorTurma(turma);

Depois não esqueça de colocar esse command lá quando cria uma atividade avaliativa.